lib/src/widgets/setup_game_code.dart

```diff
@@ -33,6 +33,7 @@ class _SetupGameCodeWidgetState extends State<SetupGameCodeWidget> {
           child: TextFormField(
             decoration: InputDecoration(labelText: 'Game Code'),
             controller: _gameCodeController,
+            onEditingComplete: () => widget.onJoinGameClicked(_gameCodeController.text),
```

Comment:
Maybe you could rename `onJoinGameClicked` to `onJoinGame` since it no longer is just a click.